unit_tests/test_tlslite_extensions.py

```diff
@@ -1392,6 +1392,24 @@ def test_parse_with_extra_data_at_end(self):
         with self.assertRaises(SyntaxError):
             ext.parse(parser)
 
+    def test___repr__(self):
+        ext = SignatureAlgorithmsExtension().create([(HashAlgorithm.sha1,
+                                                      SignatureAlgorithm.rsa),
+                                                     (HashAlgorithm.sha256,
+                                                      SignatureAlgorithm.rsa),
+                                                     (HashAlgorithm.sha384,
+                                                      SignatureAlgorithm.dsa)])
+
+        self.assertEqual(repr(ext),
+                "SignatureAlgorithmsExtension("
+                "sigalgs=[rsa_pkcs1_sha1, rsa_pkcs1_sha256, (sha384, dsa)])")
```

Comment:
The interface is reasonable. The only concern is that an generic algorithm today e.g., rsa-sha224, will change name in the future if it becomes part of a TLS 1.3 revision. I don't know if that concern applies to the intended use of the interface (according to python docs repr() is about debugging, so most likely it does not).